### DISCUSSION TITLE
Helper methods for components?

### DISCUSSION THREAD

**Discussion by luwes** [strapi/strapi#7379] [Legacy Topics]
I think I might be missing a way to make it easier to work with components to create, update, etc.
similar to how it's easier to work with content types.

Is there a way to make it easier? Is there some way to add the `.update()`, `.find()` methods to components?

So far I've been using the low level Bookshelf API.

```js
  /**
   * Promise to find all profiles.
   * @return {Promise}
   */
  async find(params) {
    const TranscodeProfile = strapi.components['media.transcode-profile'];
    const result = await TranscodeProfile.where(params).fetchAll();
    return result.toJSON();
  },
```

The above code is still fine but I'm about to work on a step where a content type has a component which also holds another component. The nesting and relations would probably require some transactions etc. which would make everything a bit more complex. I noticed in the `update()` method of a content type this is all nicely handled internally.

--- 

edit:

Model structure: Movie > MediaFiles > TranscodeProfile
A Movie can have many media files and each media file has a transcode profile.

Looking over the answers here it seems that components might not be the correct thing for this?
https://github.com/strapi/strapi/issues/5124#issuecomment-611954293

**Comment by derrickmehaffy** [comment]
Hello, I will ping @alexandrebodin to see if he has any ideas. But yes generally when you start getting this nested, components start getting very complex and it might just be easier to handle these as relations.

**Reply by alexandrebodin** [reply]
Yeah I want to add some programmatic APIs for that taht would make a lot of sense

**Comment by luwes** [comment]
thanks guys, I changed the components into content types in the mean time.

I mostly tried out components because the nice nested admin UI they offer but it seems the internals make everything a bit more complex and limited. made me think why components are not build upon the content type model? since it is some kind of a relation under the hood.

anyway great job with Strapi! loving it, the content type builder is really cool!

**Reply by derrickmehaffy** [reply]
You are correct, it's a polymorphic relation under the hood similar to how the upload plugin works and likewise the media field.
